cmd/cosign/cli/attest/attest_blob.go

```diff
@@ -189,28 +200,69 @@ func (c *AttestBlobCommand) Exec(ctx context.Context, artifactPath string) error
 		if err != nil {
 			return err
 		}
-		var entry *models.LogEntryAnon
 		if c.RekorEntryType == "intoto" {
-			entry, err = cosign.TLogUploadInTotoAttestation(ctx, rekorClient, sig, rekorBytes)
+			rekorEntry, err = cosign.TLogUploadInTotoAttestation(ctx, rekorClient, sig, rekorBytes)
 		} else {
-			entry, err = cosign.TLogUploadDSSEEnvelope(ctx, rekorClient, sig, rekorBytes)
+			rekorEntry, err = cosign.TLogUploadDSSEEnvelope(ctx, rekorClient, sig, rekorBytes)
 		}
 
 		if err != nil {
 			return err
 		}
-		fmt.Fprintln(os.Stderr, "tlog entry created with index:", *entry.LogIndex)
-		signedPayload.Bundle = cbundle.EntryToBundle(entry)
+		fmt.Fprintln(os.Stderr, "tlog entry created with index:", *rekorEntry.LogIndex)
+		signedPayload.Bundle = cbundle.EntryToBundle(rekorEntry)
 	}
 
 	if c.BundlePath != "" {
-		signedPayload.Base64Signature = base64.StdEncoding.EncodeToString(sig)
-		signedPayload.Cert = base64.StdEncoding.EncodeToString(rekorBytes)
+		var contents []byte
+		if c.ProtobufBundleFormat {
+			// Determine if signature is certificate or not
+			var hint string
+			var rawCert []byte
+
+			signer, err := sv.Bytes(ctx)
+			if err != nil {
+				return fmt.Errorf("error getting signer: %w", err)
+			}
+			cert, err := cryptoutils.UnmarshalCertificatesFromPEM(signer)
+			if err != nil || len(cert) == 0 {
+				hashedBytes := sha256.Sum256(signer)
+				hint = base64.StdEncoding.EncodeToString(hashedBytes[:])
+			} else {
+				rawCert = cert[0].Raw
+			}
+
+			bundle, err := cbundle.MakeProtobufBundle(hint, rawCert, rekorEntry, timestampBytes)
+			if err != nil {
+				return err
+			}
+
+			bundle.Content = &protobundle.Bundle_DsseEnvelope{
+				DsseEnvelope: &protodsse.Envelope{
+					Payload:     payload,
+					PayloadType: c.PredicateType,
+					Signatures: []*protodsse.Signature{
+						&protodsse.Signature{
+							Sig: sig,
+						},
+					},
+				},
+			}
+
+			contents, err = protojson.Marshal(bundle)
+			if err != nil {
+				return err
+			}
```

Comment:
nit - consider extracting this section into its own function, it's a bit long and the `Exec` function is already extremely long.